### ISSUE TITLE
Core crashing on logging out 

### ISSUE THREAD

**Issue by Qrysto** [Nexusoft/LLL-TAO#93]
There's a bug that doesn't occur 100% of the time but I have encountered more than a few times: when I log out of my sigchain, the core just crashes (multiuser was off)

**Comment by VidereLicet** [comment]
Okay, I'll see if I can get it to happen with a stack trace and post a commit here when it is finished.

**Comment by Qrysto** [comment]
@VidereLicet There are not many strange errors in the log after the core crashes, here's the few last lines 

```
[01:18:20.133] Accept : Mempool Consistency Check Complete in 0 ms
[01:18:20.134] Tritium Node : 66.85.76.106 ACTION::NOTIFY: Synchronization COMPLETE at f8a3008902d0e591a7d6
[01:18:20.134] Tritium Node : 66.85.76.106 ACTION::NOTIFY: Synchronized 2 blocks in 0 seconds [2 blocks/s]
[01:18:30.062] StakeMinterThread : Stake Minter Started
[01:18:30.062] StakeMinterThread : Stake Minter Initialized
[01:18:54.475] Stop : Shutting down Stake Minter
```

**Comment by Qrysto** [comment]
@VidereLicet Can you reproduce the issue? Now it happens pretty consistently for me

**Comment by Qrysto** [comment]
@VidereLicet Latest error log with verbose=3:

```
[21:07:28.252] Base Node : :: Connecting to 5.254.6.16
[21:07:28.252] Attempt : Connection failed 5.254.6.16:0 (10049)
[21:07:28.291] Tritium Node : 66.85.76.106 Latency (Nonce c68922fbe9ef341f - 250 ms)
[21:07:28.695] Stop : Shutting down Stake Minter
[21:07:29.379] Base Node : :: Connecting to 5.254.6.16
[21:07:29.379] Attempt : poll timeout 5.254.6.16:9888 (10035)
```

**Comment by KenCorma** [comment]
Ok, So seems like it only happens when your unlocked for staking

**Comment by VidereLicet** [comment]
Testing staking from genesis to see if I can track down where this issue is

**Comment by Qrysto** [comment]
Yeah it really seems that it only happens when `-stake=1`

**Comment by VidereLicet** [comment]
Resolved and tested on testnet with commit 696d2c4c4e6cb15a276ccd786dc78018c34897f6
